### DISCUSSION TITLE
Add JPEG-XL Support

### DISCUSSION THREAD

**Discussion by CrushedAsian255** [spacedriveapp/spacedrive#2191] [Feature Request]
Hello! On my computer I have a lot of *.jxl (JPEG-XL) image files. These open fine in Finder, as they are built into the CoreImage framework. However, SpaceDrive doesn't seem to be able to read them, and just shows them as an "unknown" file. [A reference decoder is available.](https://github.com/libjxl/libjxl) Thanks for considering.

**Comment by brxken128** [comment]
I'm not too sure the reference decoder would be much help, as we'd likely have to build and bundle it for each of the platforms that we support, all for one specific format. If there was a Rust decoder available, we could likely read them just fine and even create thumbnails.

One thing we _can_ do is add extension and magic byte detection for `.jxl` files so they correctly display as images, but they won't have thumbnails until we can decode them sadly.

**Comment by CrushedAsian255** [comment]
> If there was a Rust decoder available, we could likely read them just fine and even create thumbnails.

There is a community-made Rust decoder available whose developer has ties with the devs of the file format
https://github.com/tirr-c/jxl-oxide

**Reply by brxken128** [reply]
I'll create a Linear issue for this and give it a shot when I get a chance!
